pkgs/by-name/dr/dragonflydb/package.nix

```
@@ -1,109 +1,275 @@
 {
   fetchFromGitHub,
-  fetchurl,
   lib,
   stdenv,
+
+  # Dependencies
+  abseil-cpp,
+  c-ares,
+  croncpp,
   double-conversion,
-  gperftools,
-  mimalloc,
-  rapidjson,
-  liburing,
-  xxHash,
+  expected-lite,
+  fast-float,
+  flatbuffers_23,
   gbenchmark,
-  glog,
+  gperftools,
   gtest,
+  hdrhistogram_c,
+  hnswlib,
   jemalloc,
-  gcc-unwrapped,
+  liburing,
+  lz4,
+  pcre2,
+  pugixml,
+  rapidjson,
+  re2,
+  re-flex,
+  uni-algo,
+  xxHash,
+  zstd,
+
+  # Build tools
   autoconf,
   autoconf-archive,
-  automake,
+  automake116x,
+  bison,
   cmake,
+  gcc-unwrapped,
   ninja,
+
+  # Runtime dependencies
   boost,
-  libunwind,
   libtool,
+  libunwind,
   openssl,
+  zlib,
+
+  # Test dependencies
+  python3,
+
+  # Options
+  withAws ? true,
+  withGcp ? true,
+  withGperf ? true,
+  withPcre ? true,
+  withRe2 ? true,
+  withSearch ? true,
+  withUnwind ? !stdenv.targetPlatform.isAarch64,
+
+  withAsan ? false,
+  withUsan ? false,
 }:
 
 let
   pname = "dragonflydb";
-  version = "0.1.0";
+  version = "1.34.2";
 
   src = fetchFromGitHub {
-    owner = pname;
+    owner = "dragonflydb";
     repo = "dragonfly";
     tag = "v${version}";
-    hash = "sha256-P6WMW/n+VezWDXGagT4B+ZYyCp8oufDV6MTrpKpLZcs=";
+    hash = "sha256-n70IB32tZDe665hVLrKC0BSSJutmYhtPJvfNa48xaqA=";
+    fetchSubmodules = true;
+  };
+
+  aws-sdk-cpp-1-11-162 = fetchFromGitHub {
+    owner = "aws";
+    repo = "aws-sdk-cpp";
+    tag = "1.11.162";
+    hash = "sha256-NxVE7H8BOetpbBkB2PTVBoHSXYm6cTp41F1LJmhtBbs=";
     fetchSubmodules = true;
   };
 
-  # Needed exactly 5.4.4 for patch to work
-  lua = fetchurl {
-    url = "https://github.com/lua/lua/archive/refs/tags/v5.4.4.tar.gz";
-    hash = "sha256-L/ibvqIqfIuRDWsAb1ukVZ7c9GiiVTfO35mI7ZD2tFA=";
+  glog-absl = fetchFromGitHub {
+    owner = "romange";
+    repo = "glog";
+    tag = "Absl";
+    hash = "sha256-68Hx3kIPgyMSdHCUpYr68Cw8V4Umtyd+4VLZc3zUb1s=";
+  };
+
+  jsoncons-dragonfly = fetchFromGitHub {
+    owner = "dragonflydb";
+    repo = "jsoncons";
+    tag = "Dragonfly.178";
+    hash = "sha256-cxM95DFFo5z+eImgZzJw+ykaeSDtBF+hw5qo6gnL53s=";
```

`Dragonfly.178` is a branch, not a tag, so this doesn't work.

```suggestion
    rev = "990b0a2077a50a62a486eac2647f0411e65b3bbe";
    hash = "sha256-cxM95DFFo5z+eImgZzJw+ykaeSDtBF+hw5qo6gnL53s=";
```